plugins/arrow-flight-rpc/src/main/java/org/opensearch/arrow/flight/bootstrap/FlightService.java

```diff
@@ -165,6 +166,7 @@ protected void doClose() {
     }
 
     private void initializeStreamManager(FlightClientManager clientManager) {
-        streamManager = null;
+        streamManager = new FlightStreamManager(() -> allocator);
```

Comment:
Please change type of the field: `StreamManager streamManager` -> `FlightStreamManager streamManager`, no need to typecast all the time since the `streamManager` is created here